<!--
	Launch for for ZED capture node.
	Requirement: run <zed_cature>/script/generate_undist_rect_lut.py with factory calibration params file
	   	for generating camera_info .yaml files and unsidt/rect LUT files
-->

<launch>
	<!-- string for ZED camera serial number -->
	<!-- TI-Internal: SN5867575.conf:  ZED1 Camera -->
	<!--              SN29788442.conf: ZED2 Camera -->
	<arg name="zed_sn_str"             default="SN29788442"/>
	<!-- to find your device name, use ls /dev/video* and look for the name begin with video -->
	<!-- typically camera recoginized as "/dev/video0" on the target -->
	<arg name="device_name"            default="/dev/video0"/>
	<!-- camera_mode: '2K' - 2208x1242, 'FHD' - 1920x1080, 'HD' - 1280x720, 'HD2' - 1280x720: center-cropped from 1080p -->
	<arg name="camera_mode"            default="HD"/>
	<!-- frame rate -->
	<arg name="frame_rate"             default="15"/>
	<!-- encoding: "yuv422" (default), "bgr8" -->
	<arg name="encoding"               default="yuv422"/>

	<node pkg="zed_capture" type="zed_capture" name="zed_capture" output="screen" required="true">
		<param name="camera_mode"            value="$(arg camera_mode)"/>
		<param name="frame_rate"             value="$(arg frame_rate)"/>
		<param name="frame_id_left"          value="left_frame"/>
		<param name="frame_id_right"         value="right_frame"/>
		<param name="device_name"            value="$(arg device_name)"/>
		<param name="encoding"               value="$(arg encoding)"/>
		<param name="camera_info_left_yaml"  value="$(arg zed_sn_str)_$(arg camera_mode)_camera_info_left.yaml"/>
		<param name="camera_info_right_yaml" value="$(arg zed_sn_str)_$(arg camera_mode)_camera_info_right.yaml"/>
	</node>

</launch>
